src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,496 @@
+  static const unsigned FREQUENCY_RANGE = MAX_FREQUENCY - MIN_FREQUENCY;
+
+  /** 
+ * max frequency id
+ * the highest frequency ID which can be set in AR62xx.
+ * this number is used to calculate the particular frequency-id of each frequency.
+ * 
+ * explanation: the AR62xx uses an integer value for each 8.33/25 kHz frequency starting at 118.000 MHz
+ * e.g. frequency-id 0    => 118.000 MHz
+ *      frequency-id 1    => 118.005 MHz
+ *      frequency-id 3040 => 137.000 MHz
+ */
+  static const int MAX_FREQUENCY_ID = 3040;
+
+  /** bitmasks to get the frequency out of a frequency-id */
+  static const int FREQUENCY_BITMASK = 0xFFF0;
+
+  /** bitmasks to get the channel out of a frequency-id */
+  static const int CHANNEL_BITMASK = 0xF;
+
+  /** ------------------------------- VARIABLES ------------------------------- */
+private:
+  /** Port the radio is connected to. */
+  Port &port;
+
+  /** active station frequency*/
+  RadioFrequency active_frequency;
+
+  /** passive (or standby) station frequency*/
+  RadioFrequency passive_frequency;
+
+  /** ------------------------------- CONSTRUCTOR ------------------------------- */
+public:
+  explicit AR62xxDevice(Port &_port);
+
+  /** ------------------------------- FUNCTIONS ------------------------------- */
+private:
+  /**
+   * @brief Send - Sends a message to the AR62xx
+   * 
+   * @param msg message
+   * @param msg_size size of the message
+   * @param env operation environment to run the thread in
+   * @return true if message has been sent
+   * @return false if message could not be sent
+   */
+  bool Send(const uint8_t *msg, size_t msg_size, OperationEnvironment &env);
+
+  /**
+   * @brief converts a frequency to an AR62xx-ID
+   * 
+   * explanation: the AR62xx uses an integer value for each 8.33/25 kHz frequency starting at 118.000 MHz
+   * e.g. frequency-id 0    => 118.000 MHz
+   *      frequency-id 1    => 118.005 MHz
+   *      frequency-id 3040 => 137.000 MHz
+   * 
+   * @param frequency frequency in MHz
+   * @return uint16_t frequency-id
+   */
+  uint16_t FrequencyToAR62xxId(const RadioFrequency frequency);
+
+  /**
+   * @brief converts an AR62xx-ID to a real frequency in MHz
+   * 
+   * @param frequency_id frequency-id
+   * @return RadioFrequency frequency in MHz
+   */
+  RadioFrequency AR62xxIdToFrequency(const uint16_t frequency_id);
+
+  /**
+   * @brief generates a command to set a station in the AR62xx and outputs the length of this command
+   * 
+   * @param command input=buffer for the command (type=uint8_t[MAX_LENGTH]), output=the command to set the station
+   * @param station_id the id of the station which should be set
+   *        1 = active station of AR62xx
+   *        0 = passive station of AR62xx
+   * @param frequency frequency in MHz to set
+   * @return unsigned command-length
+   */
+  unsigned int GenerateSetStationCommand(uint8_t *command, unsigned int station_id, const RadioFrequency frequency);
```

Comment:
because it needs for generating the command the current active AND passive frequency as for setting a frequency (passive or active) we are setting both frequencies (even if one frequency is the same afterwards)
e.g. setting active frequency
command: [new frequency of active, old frequency of passive]

Reply:
This is bad API design, and your API documentation is incomplete.
You should separate packet marshalling and property management.

Reply:
please have a look at the new implementation which is completely different from this and fully documented :-)